core/src/main/scala/akka/kafka/internal/KafkaConsumerActor.scala

```diff
@@ -0,0 +1,162 @@
+
+import java.util
+
+import akka.Done
+import akka.actor.{Actor, ActorRef, Cancellable, Status}
+import akka.event.LoggingReceive
+import akka.kafka.internal.ConsumerStage.{CommittableOffsetBatchImpl, Committer}
+import akka.kafka.scaladsl.Consumer.PartitionOffset
+import akka.util.Timeout
+import org.apache.kafka.clients.consumer._
+import org.apache.kafka.common.TopicPartition
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.duration._
+
+object KafkaConsumerActor {
+  //requests
+  case class Subscribe(topics: Set[String])
+  case class RequestMessages(topics: Set[TopicPartition])
+  object RequestAnyMessages
+  case class Commit(offsets: Map[TopicPartition, Long])
+  //responses
+  case class Assigned(partition: List[TopicPartition])
+  case class Revoked(partition: List[TopicPartition])
+  case class Messages[K, V](messages: Iterator[ConsumerRecord[K, V]])
+  case class Committed(offsets: Map[TopicPartition, OffsetAndMetadata])
+  //internal
+  private case object Poll
+}
+
+class KafkaConsumerActor[K, V](consumerFactory: () => KafkaConsumer[K, V]) extends Actor {
+  import KafkaConsumerActor._
+
+  import scala.collection.JavaConversions._
+  import scala.collection.JavaConverters._
+  import scala.concurrent.duration._
+
+  def pollTimeout() = 10.millis
+  def pollDelay() = 50.millis
+
+  private var requests = Map.empty[TopicPartition, ActorRef]
+  var consumer: KafkaConsumer[K, V] = _
+
+  def receive: Receive = LoggingReceive {
+    case Commit(offsets) =>
+      val commitMap = offsets.mapValues(new OffsetAndMetadata(_))
+      val reply = sender
+      consumer.commitAsync(commitMap, new OffsetCommitCallback {
+        override def onComplete(offsets: util.Map[TopicPartition, OffsetAndMetadata], exception: Exception): Unit = {
+          if (exception != null) reply ! Status.Failure(exception)
+          else reply ! Committed(offsets.asScala.toMap)
+        }
+      })
+      //right now we can not store commits in consumer - https://issues.apache.org/jira/browse/KAFKA-3412
+      poll(immediatePoll = false)
+
+    case Subscribe(topics) =>
+      val reply = sender
+      consumer.subscribe(topics.toList, new ConsumerRebalanceListener {
+        override def onPartitionsAssigned(partitions: util.Collection[TopicPartition]): Unit = {
+          reply ! Assigned(partitions.toList)
+        }
+
+        override def onPartitionsRevoked(partitions: util.Collection[TopicPartition]): Unit = {
+          reply ! Revoked(partitions.toList)
+        }
+      })
+    case Poll =>
+      poll()
+    case RequestMessages(topics) =>
+      requests ++= topics.map(_ -> sender()).toMap
+      poll()
+  }
+
+  override def preStart(): Unit = {
+    super.preStart()
+    consumer = consumerFactory()
+    schedulePoll()
+  }
+
+  override def postStop(): Unit = {
+    consumer.close()
+    super.postStop()
+  }
+
+  def poll(immediatePoll: Boolean = true) = {
+    scheduled.foreach(_.cancel())
```

Comment:
a new `Poll` might already be in the mailbox so it must handle receiving such "unexpected" `Poll` also, similar if the actor is restarted